### DISCUSSION TITLE
Is there a way to update the .envrc variable?

### DISCUSSION THREAD

**Discussion by kaihendry** [direnv/direnv#1136] [Q&A]
I have a:

`export TOKEN=eyJhbGciO...` in my .envrc (amongst other variables) that needs to be manually updated sadly every day.

How do people edit that .envrc? Is there some cli like `direnv set TOKEN newTokenValue` API I am missing?

**Comment by bbenne10** [comment]
My suggestion (as I mentioned in the matrix room) is to build a small script that will idempotently retrieve the token value from the store or generate a new one if needed. then use `export TOKEN=$(getToken)` inside `.envrc` to have it refresh the token value whenever you activate the environment.
